addon/components/file-browser/component.js

```diff
@@ -0,0 +1,23 @@
+import Ember from 'ember';
+import layout from './template';
+
+/* use:
+ * {{file-browser rootNode=node selectedFile=file
+ *   onClickFile=(action) onClickNode=(action)}}
+ */
+export default Ember.Component.extend({
+    layout,
+    classNames: ['file-browser'],
+
+    selectedPath: Ember.computed('selectedFile', function() {
+        let file = this.get('selectedFile');
+        if (!file) {
+            return null;
+        }
+        let pathArray = this.get('selectedFile.materializedPath').split('/');
```

Comment:
I changed the file browser so folders and providers are not links (and are unselectable). Even with a placeholder, trying to route to a provider/folder would give console errors